includes/Documents/OrderDocument.php

```diff
@@ -1477,6 +1478,59 @@ public function get_pdf() {
 		return apply_filters( 'wpo_wcpdf_get_pdf', $pdf, $this );
 	}
 
+	/**
+	 * Replace .avif images with .jpg images in HTML
+	 * @param string $html The HTML to process
+	 * 
+	 * @return string The processed HTML
+	 */
+	public function replace_avif_images( string $html ): string {
+		$replaced_images = array();
+
+		if ( ! empty( $html ) ) {
+			$dom = new \DOMDocument();
+			$dom->loadHTML( $html );
+			
+			$images = $dom->getElementsByTagName('img');
+			foreach ( $images as $image ) {
+				if ( $image->hasAttribute( 'src' ) ) {
+					$src = $image->getAttribute( 'src' );
+					
+					if ( '.avif' === substr( $src, -5 ) ) {
+						$replaced_images[$src] = substr( $src, 0, -5 ) . '.jpg';
+					}
+				}
+			}
+			
+			if ( ! empty( $replaced_images ) ) {
+				$html = strtr( $html, $replaced_images );
+				foreach ( $replaced_images as $avif_src => $jpg_src ) {
+					// Get the server path from URL
+					if ( ! file_exists( $avif_src ) ) {
```

Comment:
We have a filesystem class now to handle this: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/e8581e1c1484d8cf41e1e4817e2e70b6f322f8e3/includes/Compatibility/FileSystem.php#L168-L180